src/Nethermind/Nethermind.Serialization.Rlp/HeaderDecoder.cs

```diff
@@ -258,6 +258,11 @@ public Rlp Encode(BlockHeader? item, RlpBehaviors rlpBehaviors = RlpBehaviors.No
             return new Rlp(rlpStream.Data);
         }
 
+        public Hash256 ComputeHeaderHash(RlpStream rlpStream)
+        {
+            return Keccak.Compute(rlpStream.PeekNextItem());
+        }
+
```

Comment:
How do you know here that the next item is actually Header? Maybe better would be to have extension method on `RlpStream` called `ComputeNextItemHash` and use it in the context of header rlp.

Reply:
Good suggestion - changed it